frame-metadata/src/v15.rs

```diff
@@ -54,8 +54,17 @@ pub struct RuntimeMetadataV15 {
 	pub call_enum_ty: <PortableForm as Form>::Type,
 	/// The type of the outer `RuntimeEvent` enum.
 	pub event_enum_ty: <PortableForm as Form>::Type,
-	/// The type of the outer `RuntimeError` enum.
-	pub error_enum_ty: <PortableForm as Form>::Type,
+	/// The module error type of the
+	/// [`DispatchError::Module`](https://docs.rs/sp-runtime/24.0.0/sp_runtime/enum.DispatchError.html#variant.Module) variant.
+	///
+	/// This provides useful information for decoding the `ModuleError` and has a similar
+	/// shape with `call_enum_ty` and `event_enum_ty`.
```

Comment:
I'm wondering about somehting like:

```suggestion
	/// The `Module` variant will be 5 scale encoded bytes which are normally decoded into 
	/// an `{ index: u8, error: [u8; 4] }` struct. This type ID points to an enum type which instead
	/// interprets the first `index` byte as a pallet variant, and the remaining `error` bytes as a 
	/// `pallet::Error` type. It is an equally valid way to decode the error bytes, and can be more
	/// informative than the usual target.
```